### DISCUSSION TITLE
Is it possible to optimize multidomain case?

### DISCUSSION THREAD

**Discussion by z-g-h** [mdolab/dafoam#641] [General]
Dear all, 
Is it possible to optimize multidomain case (For example, multistage blade optimization) ? If not , is it possible to add this feature or how to add this feature to code?

**Comment by friedenhe** [comment]
The current version does not support the rotor-stator interface BC. However, this is something we want to add. We do not have a timeline to do this, though....

**Reply by z-g-h** [reply]
Dear Professor,
 Thanks for your applying. Recently, I want to develop code to support mixingPlane Interface BC(similar to of-extend version), so I checked coloring code about couple BC. It seems that it can't coloring AMI BC correctly in function "DAJacCon::calcNeiBFaceGlobalCompact(labelList& neiBFaceGlobalCompact)", so is that means  there will be a large change for dafoam to support mixingPlane BC ?
If there is a large changes, could you please give some suggestions where should be changed except for BC code itself?
Best wishes

**Comment by friedenhe** [comment]
You are right, DAFoam does not support AMI BC at this moment. Are you trying to do steady-state or unsteady simulations for multi-state turbomachinery? As I understand, the steady-state mixingPlane BC does not need AMI. It would be great if you could add the mixingPlane BC to OpenFOAM/DAFoam.

**Reply by z-g-h** [reply]
Rotor/stator interface pitch  areas are often different, so steady-state mixingPlane need a interpolation between interfaces which is similar to AMI  BC to achieve circumferential average and no-matching mesh interpolation can't be avoid in mixingPlane interface. I will try to turn off the coloring acceleration to directly use a brute-force perturbation to get correct Jacobian matrix firstly.

btw, I find that you say AD for AMI BC in single processor is ok ([#466](https://github.com/mdolab/dafoam/discussions/466)), is that means I can use AD to deal with interface which is similar to AMI BC?

**Comment by friedenhe** [comment]
I guess foam-extend's mixing plane implementation does depend on AMI. But you may be able to find other mixing-plane BC implementations that do not depend on AMI. If you want to go with the AMI option, you can force the rotor/stator interface patch to be owned by the same processor, and DAFoam should be able to handle that. Do not turn off the coloring; the preconditioner calculation does not need to be very accurate, so it may be able to handle AMI.
